astra/src/main/java/com/slack/astra/clusterManager/ClusterHpaMetricService.java

```diff
@@ -160,15 +211,38 @@ protected static double calculateDemandFactor(
     return Math.ceil(rawDemandFactor * 100) / 100;
   }
 
+  private static double calculateDemandFactorFromCacheNodeCapacity(
+      long totalBytesRequiringAssignment, long totalCacheNodeCapacityBytes) {
+    if (totalCacheNodeCapacityBytes == 0) {
+      LOG.error("No cache node capacity is detected");
+      return 1;
+    }
+
+    double rawDemandFactor = (double) totalBytesRequiringAssignment / totalCacheNodeCapacityBytes;
+    LOG.info(
+        "Calculating demand factor from ng cache nodes: bytes needed: {}, capacity: {}, demandFactor: {}",
+        totalBytesRequiringAssignment,
+        totalCacheNodeCapacityBytes,
+        Math.ceil(rawDemandFactor * 100) / 100);
+    return Math.ceil(rawDemandFactor * 100) / 100;
+  }
+
   /** Updates or inserts an (ephemeral) HPA metric for the cache nodes. This is NOT threadsafe. */
-  private void persistCacheConfig(String replicaSet, Double demandFactor) {
+  private synchronized void persistCacheConfig(String replicaSet, Double demandFactor) {
```

Comment:
Why do you use both synchronized and a reentrant lock here? The latter is generally a pattern to use now, in case something was ever refactored in the future and we wanted to run this in a virtual thread.

Reply:
Heh this was a remnant of my attempts to debug the manager crashing. I've removed it for now!